core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -347,3 +350,77 @@ object InvokerActor {
 
   val timerName = "testActionTimer"
 }
+
+/*
+ * Actor to handle invoker registration
+ */
+class Registrar(sendIdToInvoker: (InvokerIdResponseMessage) => Future[RecordMetadata],
+                registerConsumer: MessageConsumer,
+                redisClient: RedisClient)
+    extends Actor {
+
+  implicit val transid = TransactionId.registrar
+  implicit val logging = new AkkaLogging(context.system.log)
+  implicit val timeout = Timeout(5.seconds)
+  implicit val ec = context.dispatcher
+
+  def receive = {
+    case m: InvokerIdRequestMessage => {
+      val assignedId =
+        if (m.proposedId != -1) {
+          // static id assignment prototcol -- always accept proposedId
+          logging.info(this, s"invoker with name ${m.instanceName} was assigned its proposedId ${m.proposedId}")
+          m.proposedId
+        } else {
+          // dynamic id assignment protocol
+          val priorId = redisClient.hget("controller:registar:idAssignments", m.instanceName)
```

Comment:
It does seem like the redis client you're using has a blocking interface. If that's correct, this is quite dangerous, as threads will be blocked on our main dispatcher effectively reducing throughput while a registration is in progress.

It doesn't seem like we need an Actor in this case either as no state is encapsulated. Could the whole registration be handled by the message consumer then?

Reply:
Good point.  When I changed things to externalize the state to Redis I could have eliminated the Actor.  I'll do that.